pkg/chart/chart.go

```diff
@@ -314,8 +314,13 @@ func (t *Testing) processCharts(action func(chart *Chart) TestResult) ([]TestRes
 		if err != nil {
 			return results, errors.Wrap(err, "Error identifying merge base")
 		}
-		t.git.AddWorkingTree(ctPreviousRevisionTree, mergeBase)
-		defer t.git.RemoveWorkingTree(ctPreviousRevisionTree)
+		// Remove worktree first (it may not have been cleaned up correctly from the last test run)
+		// TODO: This may no longer be needed once https://github.com/helm/chart-testing/issues/60
+		//       is addressed.
+		t.git.RemoveWorkTree(ctPreviousRevisionTree)
```

Comment:
There's a (very small) chance that we could be deleting an actively used worktree here. Currently our worktree is created at the path `ct_previous_revision`, and this seems fine to me.

To be _extra_ safe we could set the worktree path to something even more clearly owned by helm, like `helm.sh/ct_previous_revision`. Thoughts?

Reply:
What about creating a temp dir? Something like this:

`ioutil.TempFile("helm.sh", "ct_previous_revision")`

Reply:
@unguiculus it does seem like this would be a great approach, but we'd probably need to clean up the log formatting. I tried using `ioutil.TempDir` and on mac it makes the logs much less user-friendly:

```
Installing chart '/var/folders/09/t_s1mmqd3ls74fd5b6h4hg0h0000gp/T/ct_previous_revision731558328/stable/redis' with values file '/var/folders/09/t_s1mmqd3ls74fd5b6h4hg0h0000gp/T/ct_previous_revision731558328/stable/redis/ci/default-values.yaml'...
```

Maybe we should go with the current approach for now? I think it's also potentially nice to be able to debug with the git worktree at a relative path. With the `helm.sh` prefix, logs look like this:

```
Installing chart 'helm.sh/ct_previous_revision/stable/redis' with values file 'helm.sh/ct_previous_revision/stable/redis/ci/default-values.yaml'...
```

Reply:
Of course, it's `ioutil.TempDir`, not `ioutil.TempFile` as I had suggested. If you call it with the `dir` parameter, it creates the temp dir in this directory. Only if you pass an empty string, it delegates to `os.TempDir` to compute the directory.

Reply:
Okay so I did a little more experimentation. In order for `ioutil.TempDir` to work inside `helm.sh`, `helm.sh` has to already exist. We could create it first but that's kind of a mess since we'd need to do some bookkeeping to make sure we never delete `helm.sh` if it already existed or its contents changed outside of `ct`.

Second approach: Just generate a worktree path at the top level of the repo. This is much simpler, but the downside is that it's a tiny bit ugly:

```
Installing chart 'ct_previous_revision223934316/stable/redis' with values file 'ct_previous_revision223934316/stable/redis/ci/default-values.yaml'...
```

Reply:
@unguiculus thoughts on the top-level tempdir vs `helm.sh/ct_previous_revision`?

Reply:
What about something simple like this? 
> If path is already a directory, MkdirAll does nothing and returns nil.
```diff
	if err := os.MkdirAll("helm.sh", 0755); err != nil {
		fmt.Println(errors.Wrapf(err, "Error creating worktree directory for previous revision of chart '%s'\n", chart))
	}
```

Reply:
I'm fine with top-level or @scottrigby's solution.

Reply:
Same

Reply:
Cool, I'll push the top-level changes up then. `os.MkdirAll` was what I was alluding to in my [above comment](https://github.com/helm/chart-testing/pull/133#discussion_r268778090) -- we run into the same problems if we want to avoid creating side effects as a result of running `ct`.